src/mame/tecmo/spbactn.cpp

```diff
@@ -783,11 +808,26 @@ void spbactnp_state::spbactnp(machine_config &config)
 	Z80(config, m_audiocpu, XTAL(4'000'000));
 	m_audiocpu->set_addrmap(AS_PROGRAM, &spbactnp_state::sound_map);
 
-	// yes another CPU..
-	z80_device &extracpu(Z80(config, "extracpu", XTAL(4'000'000)));
-	extracpu.set_addrmap(AS_PROGRAM, &spbactnp_state::extra_map);
-	extracpu.set_vblank_int("screen", FUNC(spbactnp_state::irq0_line_hold));
-//  extracpu.set_vblank_int("screen", FUNC(spbactnp_state::nmi_line_pulse));
+	GENERIC_LATCH_16(config, m_extralatch);
+	m_extralatch->data_pending_callback().set_inputline(m_extracpu, INPUT_LINE_NMI);
+
+	GFXDECODE(config, m_extragfxdecode, m_extrapalette, gfx_extraspbactnp);
+
+	SCREEN(config, m_extrascreen, SCREEN_TYPE_RASTER);
+	m_extrascreen->set_refresh_hz(60);
+	m_extrascreen->set_vblank_time(ATTOSECONDS_IN_USEC(0));
+	m_extrascreen->set_size(32*8, 32*8);
+	m_extrascreen->set_visarea(0*8, 32*8-1, 2*8, 30*8-1);
+	m_extrascreen->set_screen_update(FUNC(spbactnp_state::extrascreen_update));
+
+	config.set_default_layout(layout_spbactnp);
+
+	Z80(config, m_extracpu, XTAL(4'000'000));
+	m_extracpu->set_addrmap(AS_PROGRAM, &spbactnp_state::extra_map);
+	m_extracpu->set_vblank_int("extrascreen", FUNC(spbactnp_state::irq0_line_hold));
+
+	PALETTE(config, m_extrapalette).set_format(palette_device::xBRG_444, 0x1000 / 2);
+	m_extrapalette->set_endianness(ENDIANNESS_BIG);
 
 	// video hardware
 	SCREEN(config, m_screen, SCREEN_TYPE_RASTER);
```

Comment:
The headings are messed up now, since the video hardware etc. are intermixed.

Reply:
Ok I'll take care of it